### PR TITLE
Ntp

### PR DESCRIPTION
**Pull request by GanShun**
Simple ntp client to set time.

Nichrome needs this or else chrome keeps complaining when you don't have a reasonable date.

### CODE REVIEW

File: cmds/ntpd/ntpd.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+		if t, err = ntp.Time(s); err == nil {
+			// Right now we return on the first valid time.
+			// We can implement better heuristics here.
+			debug("Got time %v\n", t)
+			return
+		}
+		debug("Error getting time: %v\n", err)
+	}
+
+	t = time.Now()
```
Comment (rminnich) [inline cmds/ntpd/ntpd.go]: why bother with this extra work? t will have the zero value. You should never use it in the case of an error.

Comment (rminnich) [reply cmds/ntpd/ntpd.go]: you could also have it return t *time.Time and then return nil, err if it's an error?

File: cmds/ntpd/ntpd.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	}
+
+	debug("Reading NTP servers from config file: %v\n", *config)
+	f, err := os.Open(*config)
+	if err == nil {
+		defer f.Close()
+		servers = parseServers(bufio.NewReader(f))
+		debug("Found %v servers \n", len(servers))
+	} else {
+		debug("Unable to open config file: %v\n", err)
```
Comment (rminnich) [inline cmds/ntpd/ntpd.go]: this is an error, maybe it's worth printing it alwys? Why not one print now two?

Comment (GanShun) [reply cmds/ntpd/ntpd.go]: Hmm ok I'll do that

File: cmds/ntpd/ntpd.go (hunk 1 above)
Comment (GanShun) [reply cmds/ntpd/ntpd.go]: I'll change it to return nil

File: cmds/ntpd/ntpd.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+	}
+
+	debug("Reading NTP servers from config file: %v\n", *config)
+	f, err := os.Open(*config)
+	if err == nil {
+		defer f.Close()
+		servers = parseServers(bufio.NewReader(f))
+		debug("Found %v servers \n", len(servers))
+	} else {
+		log.Printf("Unable to open config file: %v\n", err)
```
Comment (rminnich) [inline cmds/ntpd/ntpd.go]: just seems weird to me, still, that this is two lines of printf, not one? Also for log.Printf you don't need a \n, it tends to add one for you IIRC.

Comment (GanShun) [reply cmds/ntpd/ntpd.go]: oh ok. And I kinda want to turn it into a daemon at some point, but I don't have a time. i can change the name if you really want

Comment (GanShun) [reply cmds/ntpd/ntpd.go]: Done

**Pull request merged**